LiveSched/src/main/java/dev/coms4156/project/livesched/Task.java

```diff
@@ -149,11 +149,24 @@ public void updateResource(ResourceType resourceType, int quantity) {
     if (quantity < 0) {
       throw new IllegalArgumentException("Quantity cannot be negative.");
     }
-    
-    if (resourceList.containsKey(resourceType)) {
-      resourceList.replace(resourceType, quantity);
+
+    // Find an existing ResourceType with the same typeName
+    ResourceType existingType = null;
+    for (ResourceType type : resourceList.keySet()) {
```

Comment:
Instead of checking in a for loop, would it be more efficient to check `quantity == 0` in containsKey condition?

Reply:
ex. If contains (resource type exists in list) and quantity == 0, remove else (quantity !=0) replace else (resource type doesn't exist) add

Reply:
Oh the issue was that before when we used containsKey, somehow it didn't recognize that a resourceType already exists in the list. So, when Nurse: 2 already existed and you update Nurse to 5, it just created another Nurse:5 in the list. I thought about overriding the equals method in resourceType to compare the typeName and not the address of the object, but I figured this would be a quick fix for now!

Reply:
Hmm okay maybe it's bc we're comparing objects. Sounds good this should be good!